mmv1/third_party/terraform/services/redis/resource_redis_cluster_test.go.erb

```diff
@@ -10,6 +10,34 @@ import (
 	"github.com/hashicorp/terraform-provider-google/google/acctest"
 )
 
+func TestAccRedisCluster_createClusterWithNodeType(t *testing.T) {
+        t.Parallel()
+
+        name := fmt.Sprintf("tf-test-%d", acctest.RandInt(t))
+
+        acctest.VcrTest(t, resource.TestCase{
+                PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+                ProtoV5ProviderFactories: acctest.ProtoV5ProviderBetaFactories(t),
+                CheckDestroy:             testAccCheckRedisClusterDestroyProducer(t),
+                Steps: []resource.TestStep{
+                        {
+                                // create cluster with replica count 1
+                                Config: createOrUpdateRedisCluster(name /* replicaCount = */, 1 /* shardCount = */, 3, true, /*nodeType = */ "REDIS_STANDARD_SMALL"),
```

Comment:
the test is throwing a prevent_destroy error here. maybe setting it to false here would fix? Does it need to be `true`?
```suggestion
                                Config: createOrUpdateRedisCluster(name /* replicaCount = */, 1 /* shardCount = */, 3, false, /*nodeType = */ "REDIS_STANDARD_SMALL"),
```

Reply:
This is similar to the current tests now. We are keeping prevent destroy enabled while creating/updating the cluster and disable it in the delete call

Reply:
This is to ensure that when terraform find this extra field in the existing cluster during update, it doesn't accidently delete the resources